src/Compilers/CSharp/Portable/Binder/BinderFactory.BinderFactoryVisitor.cs

```diff
@@ -684,6 +684,12 @@ private Binder VisitTypeDeclarationCore(TypeDeclarationSyntax parent)
                 {
                     extraInfo = NodeUsage.NamedTypeBodyOrTypeParameters;
                 }
+                else if (parent.SemicolonToken != default &&
+                    parent is RecordDeclarationSyntax &&
+                    LookupPosition.IsBetweenTokens(_position, parent.GetFirstToken(), parent.SemicolonToken.GetNextToken()))
+                {
+                    extraInfo = NodeUsage.NamedTypeBodyOrTypeParameters;
+                }
```

Comment:
It looks like this fixes the cref binding but breaks the whole records feature 😄 
I'll revert this and file a follow-up for this case.